### ISSUE TITLE
Create flexible multilevel menu widget

### ISSUE THREAD

**Issue by Jura** [undp/design-system#1098]
Split from #916 for better visibility.

Final Figma implementation: https://www.figma.com/file/WJSRV3CbcHuVWRQApwvWIm/branch/frhbJ3d1OQ03VTJtbA1PJU/UNDP-Design-System?node-id=41%3A448&t=PyY6dC0DI7gBzGWo-1

Main features:
- must accommodate arbitrary number of top level items
- when width of the nav items exceeds available width of container red ellipsis should be added after the last top level element which fits 
- clicking on ellipsis should show the rest of items pushed to the next level, underneath the top line pushing down the rest of the page content (@theadamparker , please clarify if it should indeed push or overlay the content)
- second level items should appear as dropdown under respective item, third and further levels as additional dropdowns on the side of originating item
- cascading dropdowns should always fit the viewport, hence observing on which side of the parent dropdown to appear
- use generic scripting which identify nesting by markup (via nested HTML lists and/or data attributes). There should be no or just bare minimum or initialization parameters passed to the script. All possible options should be picked from data- attributes.
- small screen styling should follow the same approach as current mega menu widget
- final implementation must be compliant with WCAG 2.1 AA requirements (proper keyboard navigation, semantic hierarchy, no cursor traps, etc)

**Comment by paboden** [comment]
@Jura @theadamparker I create a PR to start testing and debugging. It's not ready for merging at all, as there are still some issues and oddities to the setup that I'm working thru, but you can view the current state of dropdowns and overflow menus here - 
https://1098-941-flexible-menu-and-d.design-system.pages.dev/
https://github.com/undp/design-system/pull/1149
I'm working thru a list of bugs and todo's for it still (like accessibility issues, and some broken triggers and width calculations), but feel free to take a look and confirm it's headed in the right direction.

**Comment by theadamparker** [comment]
@paboden, this is a really good start. A few more points to help guide you here:

## Overflow behavior

- Building on what Jura said, a primary goal of this enhancement is to accommodate any number of menu items. These menu items should fill the available space, rather than displaying a fixed number of items. A browser 3000px wide might be able to accommodate 15 menu items, then any additional should be relegated to the overflow menu. Currently I see a glimpse of this behavior moving from tablet breakpoint to larger screen, but of course it should developed further.

## UI Adjustments

- _Language / Location / Search_ area  shouldn’t interfere with the first-level menu items
- Mobile nav items should match the full-size nav items

## Multi-level dropdown

- Menu items should accommodate long titles. Text should wrap and be left aligned. Chevron should be vertically centered. 
- The chevron will most commonly be used, but it should be built in a way where another icon (like the external icon) can be swapped in without issue.
- Frosted overlay effect should be applied to all menu levels. Right now only the second-level items should the blurred background, not the third-level items. 
- Needs to be keyboard accessible.

Really exciting to see this taking shape!

**Comment by Jura** [comment]
@paboden , can you provide instructions for developers how to construct such a menu? What are the required attributes, building blocks for top level, dropdown, etc? We need full documentation for the final product, but it also required for proper testing - I'll need to recreate the menu outside of storybook to see how it works with the real pages

**Comment by paboden** [comment]
Hi @theadamparker and @Jura - I think I was able to solve the last issues. 
The iframe mode wasn't loading all the necessary css for whatever reason. 
The RTL was a difference in some padding in the css. 
The Overflow button issue not working on your Codepen showed to me that I had to tweak the way the jQuery for the trigger button was written.
I was able to create working code pens for each with the last modifications I made:
-- Mega menu -- https://codepen.io/paboden/pen/MWPNWGw
-- Dropdown menu -- https://codepen.io/paboden/pen/GRYVJZZ
